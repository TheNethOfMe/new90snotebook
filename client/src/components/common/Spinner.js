import React from "react";

export default function Spinner() {
  return (
    <div>
      <i className="fas fa-compact-disc fa-spin" />
    </div>
  );
}
