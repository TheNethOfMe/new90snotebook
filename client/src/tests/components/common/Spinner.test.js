import React from "react";
import { shallow } from "enzyme";
import Spinner from "../../../components/common/Spinner";

let wrapper;

test("should render the Spinner", () => {
  wrapper = shallow(<Spinner />);
  expect(wrapper).toMatchSnapshot();
});
