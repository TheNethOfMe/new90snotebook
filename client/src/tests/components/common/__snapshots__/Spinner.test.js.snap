// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Spinner 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="fas fa-compact-disc fa-spin"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fas fa-compact-disc fa-spin",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="fas fa-compact-disc fa-spin"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fas fa-compact-disc fa-spin",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
